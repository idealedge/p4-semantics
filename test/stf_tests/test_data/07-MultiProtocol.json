{
  "program" : "test_data/07-MultiProtocol.p4",
  "__meta__" : {
    "version" : [2, 7],
    "compiler" : "https://github.com/p4lang/p4c"
  },
  "header_types" : [
    {
      "name" : "scalars_0",
      "id" : 0,
      "fields" : []
    },
    {
      "name" : "ethernet_t",
      "id" : 1,
      "fields" : [
        ["dstAddr", 48, false],
        ["srcAddr", 48, false],
        ["etherType", 16, false]
      ]
    },
    {
      "name" : "icmp_t",
      "id" : 2,
      "fields" : [
        ["typeCode", 16, false],
        ["hdrChecksum", 16, false]
      ]
    },
    {
      "name" : "ipv4_t",
      "id" : 3,
      "fields" : [
        ["version", 4, false],
        ["ihl", 4, false],
        ["diffserv", 8, false],
        ["totalLen", 16, false],
        ["identification", 16, false],
        ["flags", 3, false],
        ["fragOffset", 13, false],
        ["ttl", 8, false],
        ["protocol", 8, false],
        ["hdrChecksum", 16, false],
        ["srcAddr", 32, false],
        ["dstAddr", 32, false]
      ]
    },
    {
      "name" : "ipv6_t",
      "id" : 4,
      "fields" : [
        ["version", 4, false],
        ["trafficClass", 8, false],
        ["flowLabel", 20, false],
        ["payloadLen", 16, false],
        ["nextHdr", 8, false],
        ["hopLimit", 8, false],
        ["srcAddr", 128, false],
        ["dstAddr", 128, false]
      ]
    },
    {
      "name" : "tcp_t",
      "id" : 5,
      "fields" : [
        ["srcPort", 16, false],
        ["dstPort", 16, false],
        ["seqNo", 32, false],
        ["ackNo", 32, false],
        ["dataOffset", 4, false],
        ["res", 4, false],
        ["flags", 8, false],
        ["window", 16, false],
        ["checksum", 16, false],
        ["urgentPtr", 16, false]
      ]
    },
    {
      "name" : "udp_t",
      "id" : 6,
      "fields" : [
        ["srcPort", 16, false],
        ["dstPort", 16, false],
        ["length_", 16, false],
        ["checksum", 16, false]
      ]
    },
    {
      "name" : "vlan_tag_t",
      "id" : 7,
      "fields" : [
        ["pcp", 3, false],
        ["cfi", 1, false],
        ["vid", 12, false],
        ["etherType", 16, false]
      ]
    },
    {
      "name" : "ingress_metadata_t",
      "id" : 8,
      "fields" : [
        ["drop", 1, false],
        ["egress_port", 9, false],
        ["packet_type", 4, false],
        ["_padding", 2, false]
      ]
    },
    {
      "name" : "standard_metadata",
      "id" : 9,
      "fields" : [
        ["ingress_port", 9, false],
        ["egress_spec", 9, false],
        ["egress_port", 9, false],
        ["clone_spec", 32, false],
        ["instance_type", 32, false],
        ["drop", 1, false],
        ["recirculate_port", 16, false],
        ["packet_length", 32, false],
        ["enq_timestamp", 32, false],
        ["enq_qdepth", 19, false],
        ["deq_timedelta", 32, false],
        ["deq_qdepth", 19, false],
        ["ingress_global_timestamp", 48, false],
        ["lf_field_list", 32, false],
        ["mcast_grp", 16, false],
        ["resubmit_flag", 1, false],
        ["egress_rid", 16, false],
        ["checksum_error", 1, false],
        ["_padding_0", 4, false]
      ]
    }
  ],
  "headers" : [
    {
      "name" : "scalars",
      "id" : 0,
      "header_type" : "scalars_0",
      "metadata" : true,
      "pi_omit" : true
    },
    {
      "name" : "standard_metadata",
      "id" : 1,
      "header_type" : "standard_metadata",
      "metadata" : true,
      "pi_omit" : true
    },
    {
      "name" : "ethernet",
      "id" : 2,
      "header_type" : "ethernet_t",
      "metadata" : false,
      "pi_omit" : true
    },
    {
      "name" : "icmp",
      "id" : 3,
      "header_type" : "icmp_t",
      "metadata" : false,
      "pi_omit" : true
    },
    {
      "name" : "ipv4",
      "id" : 4,
      "header_type" : "ipv4_t",
      "metadata" : false,
      "pi_omit" : true
    },
    {
      "name" : "ipv6",
      "id" : 5,
      "header_type" : "ipv6_t",
      "metadata" : false,
      "pi_omit" : true
    },
    {
      "name" : "tcp",
      "id" : 6,
      "header_type" : "tcp_t",
      "metadata" : false,
      "pi_omit" : true
    },
    {
      "name" : "udp",
      "id" : 7,
      "header_type" : "udp_t",
      "metadata" : false,
      "pi_omit" : true
    },
    {
      "name" : "vlan_tag",
      "id" : 8,
      "header_type" : "vlan_tag_t",
      "metadata" : false,
      "pi_omit" : true
    },
    {
      "name" : "ing_metadata",
      "id" : 9,
      "header_type" : "ingress_metadata_t",
      "metadata" : true,
      "pi_omit" : true
    }
  ],
  "header_stacks" : [],
  "header_union_types" : [],
  "header_unions" : [],
  "header_union_stacks" : [],
  "field_lists" : [],
  "errors" : [
    ["NoError", 1],
    ["PacketTooShort", 2],
    ["NoMatch", 3],
    ["StackOutOfBounds", 4],
    ["HeaderTooShort", 5],
    ["ParserTimeout", 6]
  ],
  "enums" : [],
  "parsers" : [
    {
      "name" : "parser",
      "id" : 0,
      "init_state" : "start",
      "parse_states" : [
        {
          "name" : "parse_icmp",
          "id" : 0,
          "parser_ops" : [
            {
              "parameters" : [
                {
                  "type" : "regular",
                  "value" : "icmp"
                }
              ],
              "op" : "extract"
            }
          ],
          "transitions" : [
            {
              "value" : "default",
              "mask" : null,
              "next_state" : null
            }
          ],
          "transition_key" : []
        },
        {
          "name" : "parse_ipv4",
          "id" : 1,
          "parser_ops" : [
            {
              "parameters" : [
                {
                  "type" : "regular",
                  "value" : "ipv4"
                }
              ],
              "op" : "extract"
            }
          ],
          "transitions" : [
            {
              "value" : "0x00000501",
              "mask" : "0x00000fff",
              "next_state" : "parse_icmp"
            },
            {
              "value" : "0x00000506",
              "mask" : "0x00000fff",
              "next_state" : "parse_tcp"
            },
            {
              "value" : "0x00000511",
              "mask" : "0x00000fff",
              "next_state" : "parse_udp"
            },
            {
              "value" : "default",
              "mask" : null,
              "next_state" : null
            }
          ],
          "transition_key" : [
            {
              "type" : "field",
              "value" : ["ipv4", "fragOffset"]
            },
            {
              "type" : "field",
              "value" : ["ipv4", "ihl"]
            },
            {
              "type" : "field",
              "value" : ["ipv4", "protocol"]
            }
          ]
        },
        {
          "name" : "parse_ipv6",
          "id" : 2,
          "parser_ops" : [
            {
              "parameters" : [
                {
                  "type" : "regular",
                  "value" : "ipv6"
                }
              ],
              "op" : "extract"
            }
          ],
          "transitions" : [
            {
              "value" : "0x01",
              "mask" : null,
              "next_state" : "parse_icmp"
            },
            {
              "value" : "0x06",
              "mask" : null,
              "next_state" : "parse_tcp"
            },
            {
              "value" : "0x11",
              "mask" : null,
              "next_state" : "parse_udp"
            },
            {
              "value" : "default",
              "mask" : null,
              "next_state" : null
            }
          ],
          "transition_key" : [
            {
              "type" : "field",
              "value" : ["ipv6", "nextHdr"]
            }
          ]
        },
        {
          "name" : "parse_tcp",
          "id" : 3,
          "parser_ops" : [
            {
              "parameters" : [
                {
                  "type" : "regular",
                  "value" : "tcp"
                }
              ],
              "op" : "extract"
            }
          ],
          "transitions" : [
            {
              "value" : "default",
              "mask" : null,
              "next_state" : null
            }
          ],
          "transition_key" : []
        },
        {
          "name" : "parse_udp",
          "id" : 4,
          "parser_ops" : [
            {
              "parameters" : [
                {
                  "type" : "regular",
                  "value" : "udp"
                }
              ],
              "op" : "extract"
            }
          ],
          "transitions" : [
            {
              "value" : "default",
              "mask" : null,
              "next_state" : null
            }
          ],
          "transition_key" : []
        },
        {
          "name" : "parse_vlan_tag",
          "id" : 5,
          "parser_ops" : [
            {
              "parameters" : [
                {
                  "type" : "regular",
                  "value" : "vlan_tag"
                }
              ],
              "op" : "extract"
            }
          ],
          "transitions" : [
            {
              "value" : "0x0800",
              "mask" : null,
              "next_state" : "parse_ipv4"
            },
            {
              "value" : "0x86dd",
              "mask" : null,
              "next_state" : "parse_ipv6"
            },
            {
              "value" : "default",
              "mask" : null,
              "next_state" : null
            }
          ],
          "transition_key" : [
            {
              "type" : "field",
              "value" : ["vlan_tag", "etherType"]
            }
          ]
        },
        {
          "name" : "start",
          "id" : 6,
          "parser_ops" : [
            {
              "parameters" : [
                {
                  "type" : "regular",
                  "value" : "ethernet"
                }
              ],
              "op" : "extract"
            }
          ],
          "transitions" : [
            {
              "value" : "0x8100",
              "mask" : null,
              "next_state" : "parse_vlan_tag"
            },
            {
              "value" : "0x9100",
              "mask" : null,
              "next_state" : "parse_vlan_tag"
            },
            {
              "value" : "0x0800",
              "mask" : null,
              "next_state" : "parse_ipv4"
            },
            {
              "value" : "0x86dd",
              "mask" : null,
              "next_state" : "parse_ipv6"
            },
            {
              "value" : "default",
              "mask" : null,
              "next_state" : null
            }
          ],
          "transition_key" : [
            {
              "type" : "field",
              "value" : ["ethernet", "etherType"]
            }
          ]
        }
      ]
    }
  ],
  "deparsers" : [
    {
      "name" : "deparser",
      "id" : 0,
      "order" : ["ethernet", "vlan_tag", "ipv6", "ipv4", "udp", "tcp", "icmp"]
    }
  ],
  "meter_arrays" : [],
  "counter_arrays" : [],
  "register_arrays" : [],
  "calculations" : [],
  "learn_lists" : [],
  "actions" : [
    {
      "name" : "NoAction",
      "id" : 0,
      "runtime_data" : [],
      "primitives" : []
    },
    {
      "name" : "NoAction",
      "id" : 1,
      "runtime_data" : [],
      "primitives" : []
    },
    {
      "name" : "NoAction",
      "id" : 2,
      "runtime_data" : [],
      "primitives" : []
    },
    {
      "name" : "NoAction",
      "id" : 3,
      "runtime_data" : [],
      "primitives" : []
    },
    {
      "name" : "NoAction",
      "id" : 4,
      "runtime_data" : [],
      "primitives" : []
    },
    {
      "name" : "NoAction",
      "id" : 5,
      "runtime_data" : [],
      "primitives" : []
    },
    {
      "name" : "NoAction",
      "id" : 6,
      "runtime_data" : [],
      "primitives" : []
    },
    {
      "name" : "NoAction",
      "id" : 7,
      "runtime_data" : [],
      "primitives" : []
    },
    {
      "name" : "l2_packet",
      "id" : 8,
      "runtime_data" : [],
      "primitives" : [
        {
          "op" : "assign",
          "parameters" : [
            {
              "type" : "field",
              "value" : ["ing_metadata", "packet_type"]
            },
            {
              "type" : "hexstr",
              "value" : "0x00"
            }
          ],
          "source_info" : {
            "filename" : "test_data/07-MultiProtocol.p4",
            "line" : 179,
            "column" : 4,
            "source_fragment" : "modify_field(ing_metadata.packet_type, 0)"
          }
        }
      ]
    },
    {
      "name" : "ipv4_packet",
      "id" : 9,
      "runtime_data" : [],
      "primitives" : [
        {
          "op" : "assign",
          "parameters" : [
            {
              "type" : "field",
              "value" : ["ing_metadata", "packet_type"]
            },
            {
              "type" : "hexstr",
              "value" : "0x01"
            }
          ],
          "source_info" : {
            "filename" : "test_data/07-MultiProtocol.p4",
            "line" : 183,
            "column" : 4,
            "source_fragment" : "modify_field(ing_metadata.packet_type, 1)"
          }
        }
      ]
    },
    {
      "name" : "ipv6_packet",
      "id" : 10,
      "runtime_data" : [],
      "primitives" : [
        {
          "op" : "assign",
          "parameters" : [
            {
              "type" : "field",
              "value" : ["ing_metadata", "packet_type"]
            },
            {
              "type" : "hexstr",
              "value" : "0x02"
            }
          ],
          "source_info" : {
            "filename" : "test_data/07-MultiProtocol.p4",
            "line" : 187,
            "column" : 4,
            "source_fragment" : "modify_field(ing_metadata.packet_type, 2)"
          }
        }
      ]
    },
    {
      "name" : "mpls_packet",
      "id" : 11,
      "runtime_data" : [],
      "primitives" : [
        {
          "op" : "assign",
          "parameters" : [
            {
              "type" : "field",
              "value" : ["ing_metadata", "packet_type"]
            },
            {
              "type" : "hexstr",
              "value" : "0x03"
            }
          ],
          "source_info" : {
            "filename" : "test_data/07-MultiProtocol.p4",
            "line" : 191,
            "column" : 4,
            "source_fragment" : "modify_field(ing_metadata.packet_type, 3)"
          }
        }
      ]
    },
    {
      "name" : "mim_packet",
      "id" : 12,
      "runtime_data" : [],
      "primitives" : [
        {
          "op" : "assign",
          "parameters" : [
            {
              "type" : "field",
              "value" : ["ing_metadata", "packet_type"]
            },
            {
              "type" : "hexstr",
              "value" : "0x04"
            }
          ],
          "source_info" : {
            "filename" : "test_data/07-MultiProtocol.p4",
            "line" : 195,
            "column" : 4,
            "source_fragment" : "modify_field(ing_metadata.packet_type, 4)"
          }
        }
      ]
    },
    {
      "name" : "nop",
      "id" : 13,
      "runtime_data" : [],
      "primitives" : []
    },
    {
      "name" : "nop",
      "id" : 14,
      "runtime_data" : [],
      "primitives" : []
    },
    {
      "name" : "nop",
      "id" : 15,
      "runtime_data" : [],
      "primitives" : []
    },
    {
      "name" : "nop",
      "id" : 16,
      "runtime_data" : [],
      "primitives" : []
    },
    {
      "name" : "nop",
      "id" : 17,
      "runtime_data" : [],
      "primitives" : []
    },
    {
      "name" : "nop",
      "id" : 18,
      "runtime_data" : [],
      "primitives" : []
    },
    {
      "name" : "nop",
      "id" : 19,
      "runtime_data" : [],
      "primitives" : []
    },
    {
      "name" : "_drop",
      "id" : 20,
      "runtime_data" : [],
      "primitives" : [
        {
          "op" : "assign",
          "parameters" : [
            {
              "type" : "field",
              "value" : ["ing_metadata", "drop"]
            },
            {
              "type" : "hexstr",
              "value" : "0x01"
            }
          ],
          "source_info" : {
            "filename" : "test_data/07-MultiProtocol.p4",
            "line" : 171,
            "column" : 4,
            "source_fragment" : "modify_field(ing_metadata.drop, 1)"
          }
        }
      ]
    },
    {
      "name" : "_drop",
      "id" : 21,
      "runtime_data" : [],
      "primitives" : [
        {
          "op" : "assign",
          "parameters" : [
            {
              "type" : "field",
              "value" : ["ing_metadata", "drop"]
            },
            {
              "type" : "hexstr",
              "value" : "0x01"
            }
          ],
          "source_info" : {
            "filename" : "test_data/07-MultiProtocol.p4",
            "line" : 171,
            "column" : 4,
            "source_fragment" : "modify_field(ing_metadata.drop, 1)"
          }
        }
      ]
    },
    {
      "name" : "_drop",
      "id" : 22,
      "runtime_data" : [],
      "primitives" : [
        {
          "op" : "assign",
          "parameters" : [
            {
              "type" : "field",
              "value" : ["ing_metadata", "drop"]
            },
            {
              "type" : "hexstr",
              "value" : "0x01"
            }
          ],
          "source_info" : {
            "filename" : "test_data/07-MultiProtocol.p4",
            "line" : 171,
            "column" : 4,
            "source_fragment" : "modify_field(ing_metadata.drop, 1)"
          }
        }
      ]
    },
    {
      "name" : "set_egress_port",
      "id" : 23,
      "runtime_data" : [
        {
          "name" : "egress_port",
          "bitwidth" : 9
        }
      ],
      "primitives" : [
        {
          "op" : "assign",
          "parameters" : [
            {
              "type" : "field",
              "value" : ["ing_metadata", "egress_port"]
            },
            {
              "type" : "runtime_data",
              "value" : 0
            }
          ],
          "source_info" : {
            "filename" : "test_data/07-MultiProtocol.p4",
            "line" : 174,
            "column" : 23,
            "source_fragment" : "egress_port) { ..."
          }
        }
      ]
    },
    {
      "name" : "set_egress_port",
      "id" : 24,
      "runtime_data" : [
        {
          "name" : "egress_port",
          "bitwidth" : 9
        }
      ],
      "primitives" : [
        {
          "op" : "assign",
          "parameters" : [
            {
              "type" : "field",
              "value" : ["ing_metadata", "egress_port"]
            },
            {
              "type" : "runtime_data",
              "value" : 0
            }
          ],
          "source_info" : {
            "filename" : "test_data/07-MultiProtocol.p4",
            "line" : 174,
            "column" : 23,
            "source_fragment" : "egress_port) { ..."
          }
        }
      ]
    },
    {
      "name" : "set_egress_port",
      "id" : 25,
      "runtime_data" : [
        {
          "name" : "egress_port",
          "bitwidth" : 9
        }
      ],
      "primitives" : [
        {
          "op" : "assign",
          "parameters" : [
            {
              "type" : "field",
              "value" : ["ing_metadata", "egress_port"]
            },
            {
              "type" : "runtime_data",
              "value" : 0
            }
          ],
          "source_info" : {
            "filename" : "test_data/07-MultiProtocol.p4",
            "line" : 174,
            "column" : 23,
            "source_fragment" : "egress_port) { ..."
          }
        }
      ]
    },
    {
      "name" : "send_packet",
      "id" : 26,
      "runtime_data" : [],
      "primitives" : [
        {
          "op" : "assign",
          "parameters" : [
            {
              "type" : "field",
              "value" : ["standard_metadata", "egress_spec"]
            },
            {
              "type" : "field",
              "value" : ["ing_metadata", "egress_port"]
            }
          ],
          "source_info" : {
            "filename" : "test_data/07-MultiProtocol.p4",
            "line" : 273,
            "column" : 4,
            "source_fragment" : "modify_field(standard_metadata.egress_spec, ing_metadata.egress_port)"
          }
        }
      ]
    }
  ],
  "pipelines" : [
    {
      "name" : "ingress",
      "id" : 0,
      "init_table" : "ethertype_match",
      "tables" : [
        {
          "name" : "ethertype_match",
          "id" : 0,
          "source_info" : {
            "filename" : "test_data/07-MultiProtocol.p4",
            "line" : 198,
            "column" : 0,
            "source_fragment" : "table ethertype_match { ..."
          },
          "key" : [
            {
              "match_type" : "exact",
              "target" : ["ethernet", "etherType"],
              "mask" : null
            }
          ],
          "match_type" : "exact",
          "type" : "simple",
          "max_size" : 1024,
          "with_counters" : false,
          "support_timeout" : false,
          "direct_meters" : null,
          "action_ids" : [8, 9, 10, 11, 12, 0],
          "actions" : ["l2_packet", "ipv4_packet", "ipv6_packet", "mpls_packet", "mim_packet", "NoAction"],
          "base_default_next" : "node_7",
          "next_tables" : {
            "ipv4_packet" : "ipv4_match",
            "ipv6_packet" : "ipv6_match",
            "mpls_packet" : "ipv6_match",
            "l2_packet" : "l2_match",
            "mim_packet" : "l2_match",
            "NoAction" : "l2_match"
          },
          "default_entry" : {
            "action_id" : 0,
            "action_const" : false,
            "action_data" : [],
            "action_entry_const" : false
          }
        },
        {
          "name" : "l2_match",
          "id" : 1,
          "source_info" : {
            "filename" : "test_data/07-MultiProtocol.p4",
            "line" : 231,
            "column" : 0,
            "source_fragment" : "table l2_match { ..."
          },
          "key" : [
            {
              "match_type" : "exact",
              "target" : ["ethernet", "dstAddr"],
              "mask" : null
            }
          ],
          "match_type" : "exact",
          "type" : "simple",
          "max_size" : 1024,
          "with_counters" : false,
          "support_timeout" : false,
          "direct_meters" : null,
          "action_ids" : [16, 25, 4],
          "actions" : ["nop", "set_egress_port", "NoAction"],
          "base_default_next" : "node_7",
          "next_tables" : {
            "nop" : "node_7",
            "set_egress_port" : "node_7",
            "NoAction" : "node_7"
          },
          "default_entry" : {
            "action_id" : 4,
            "action_const" : false,
            "action_data" : [],
            "action_entry_const" : false
          }
        },
        {
          "name" : "ipv4_match",
          "id" : 2,
          "source_info" : {
            "filename" : "test_data/07-MultiProtocol.p4",
            "line" : 211,
            "column" : 0,
            "source_fragment" : "table ipv4_match { ..."
          },
          "key" : [
            {
              "match_type" : "exact",
              "target" : ["ipv4", "dstAddr"],
              "mask" : null
            }
          ],
          "match_type" : "exact",
          "type" : "simple",
          "max_size" : 1024,
          "with_counters" : false,
          "support_timeout" : false,
          "direct_meters" : null,
          "action_ids" : [14, 23, 2],
          "actions" : ["nop", "set_egress_port", "NoAction"],
          "base_default_next" : "node_7",
          "next_tables" : {
            "nop" : "node_7",
            "set_egress_port" : "node_7",
            "NoAction" : "node_7"
          },
          "default_entry" : {
            "action_id" : 2,
            "action_const" : false,
            "action_data" : [],
            "action_entry_const" : false
          }
        },
        {
          "name" : "ipv6_match",
          "id" : 3,
          "source_info" : {
            "filename" : "test_data/07-MultiProtocol.p4",
            "line" : 221,
            "column" : 0,
            "source_fragment" : "table ipv6_match { ..."
          },
          "key" : [
            {
              "match_type" : "exact",
              "target" : ["ipv6", "dstAddr"],
              "mask" : null
            }
          ],
          "match_type" : "exact",
          "type" : "simple",
          "max_size" : 1024,
          "with_counters" : false,
          "support_timeout" : false,
          "direct_meters" : null,
          "action_ids" : [15, 24, 3],
          "actions" : ["nop", "set_egress_port", "NoAction"],
          "base_default_next" : "node_7",
          "next_tables" : {
            "nop" : "node_7",
            "set_egress_port" : "node_7",
            "NoAction" : "node_7"
          },
          "default_entry" : {
            "action_id" : 3,
            "action_const" : false,
            "action_data" : [],
            "action_entry_const" : false
          }
        },
        {
          "name" : "ipv6_match",
          "id" : 4,
          "source_info" : {
            "filename" : "test_data/07-MultiProtocol.p4",
            "line" : 221,
            "column" : 0,
            "source_fragment" : "table ipv6_match { ..."
          },
          "key" : [
            {
              "match_type" : "exact",
              "target" : ["ipv6", "dstAddr"],
              "mask" : null
            }
          ],
          "match_type" : "exact",
          "type" : "simple",
          "max_size" : 1024,
          "with_counters" : false,
          "support_timeout" : false,
          "direct_meters" : null,
          "action_ids" : [15, 24, 3],
          "actions" : ["nop", "set_egress_port", "NoAction"],
          "base_default_next" : "node_7",
          "next_tables" : {
            "nop" : "node_7",
            "set_egress_port" : "node_7",
            "NoAction" : "node_7"
          },
          "default_entry" : {
            "action_id" : 3,
            "action_const" : false,
            "action_data" : [],
            "action_entry_const" : false
          }
        },
        {
          "name" : "tcp_check",
          "id" : 5,
          "source_info" : {
            "filename" : "test_data/07-MultiProtocol.p4",
            "line" : 241,
            "column" : 0,
            "source_fragment" : "table tcp_check { ..."
          },
          "key" : [
            {
              "match_type" : "exact",
              "target" : ["tcp", "dstPort"],
              "mask" : null
            }
          ],
          "match_type" : "exact",
          "type" : "simple",
          "max_size" : 1024,
          "with_counters" : false,
          "support_timeout" : false,
          "direct_meters" : null,
          "action_ids" : [18, 21, 6],
          "actions" : ["nop", "_drop", "NoAction"],
          "base_default_next" : "set_egress",
          "next_tables" : {
            "nop" : "set_egress",
            "_drop" : "set_egress",
            "NoAction" : "set_egress"
          },
          "default_entry" : {
            "action_id" : 6,
            "action_const" : false,
            "action_data" : [],
            "action_entry_const" : false
          }
        },
        {
          "name" : "udp_check",
          "id" : 6,
          "source_info" : {
            "filename" : "test_data/07-MultiProtocol.p4",
            "line" : 251,
            "column" : 0,
            "source_fragment" : "table udp_check { ..."
          },
          "key" : [
            {
              "match_type" : "exact",
              "target" : ["udp", "dstPort"],
              "mask" : null
            }
          ],
          "match_type" : "exact",
          "type" : "simple",
          "max_size" : 1024,
          "with_counters" : false,
          "support_timeout" : false,
          "direct_meters" : null,
          "action_ids" : [19, 22, 7],
          "actions" : ["nop", "_drop", "NoAction"],
          "base_default_next" : "set_egress",
          "next_tables" : {
            "nop" : "set_egress",
            "_drop" : "set_egress",
            "NoAction" : "set_egress"
          },
          "default_entry" : {
            "action_id" : 7,
            "action_const" : false,
            "action_data" : [],
            "action_entry_const" : false
          }
        },
        {
          "name" : "icmp_check",
          "id" : 7,
          "source_info" : {
            "filename" : "test_data/07-MultiProtocol.p4",
            "line" : 261,
            "column" : 0,
            "source_fragment" : "table icmp_check { ..."
          },
          "key" : [
            {
              "match_type" : "exact",
              "target" : ["icmp", "typeCode"],
              "mask" : null
            }
          ],
          "match_type" : "exact",
          "type" : "simple",
          "max_size" : 1024,
          "with_counters" : false,
          "support_timeout" : false,
          "direct_meters" : null,
          "action_ids" : [13, 20, 1],
          "actions" : ["nop", "_drop", "NoAction"],
          "base_default_next" : "set_egress",
          "next_tables" : {
            "nop" : "set_egress",
            "_drop" : "set_egress",
            "NoAction" : "set_egress"
          },
          "default_entry" : {
            "action_id" : 1,
            "action_const" : false,
            "action_data" : [],
            "action_entry_const" : false
          }
        },
        {
          "name" : "set_egress",
          "id" : 8,
          "source_info" : {
            "filename" : "test_data/07-MultiProtocol.p4",
            "line" : 276,
            "column" : 0,
            "source_fragment" : "table set_egress { ..."
          },
          "key" : [
            {
              "match_type" : "exact",
              "target" : ["ing_metadata", "drop"],
              "mask" : null
            }
          ],
          "match_type" : "exact",
          "type" : "simple",
          "max_size" : 1024,
          "with_counters" : false,
          "support_timeout" : false,
          "direct_meters" : null,
          "action_ids" : [17, 26, 5],
          "actions" : ["nop", "send_packet", "NoAction"],
          "base_default_next" : null,
          "next_tables" : {
            "nop" : null,
            "send_packet" : null,
            "NoAction" : null
          },
          "default_entry" : {
            "action_id" : 5,
            "action_const" : false,
            "action_data" : [],
            "action_entry_const" : false
          }
        }
      ],
      "action_profiles" : [],
      "conditionals" : [
        {
          "name" : "node_7",
          "id" : 0,
          "source_info" : {
            "filename" : "test_data/07-MultiProtocol.p4",
            "line" : 305,
            "column" : 8,
            "source_fragment" : "valid(tcp)"
          },
          "expression" : {
            "type" : "expression",
            "value" : {
              "op" : "==",
              "left" : {
                "type" : "field",
                "value" : ["tcp", "$valid$"]
              },
              "right" : {
                "type" : "hexstr",
                "value" : "0x01"
              }
            }
          },
          "true_next" : "tcp_check",
          "false_next" : "node_9"
        },
        {
          "name" : "node_9",
          "id" : 1,
          "source_info" : {
            "filename" : "test_data/07-MultiProtocol.p4",
            "line" : 307,
            "column" : 13,
            "source_fragment" : "valid(udp)"
          },
          "expression" : {
            "type" : "expression",
            "value" : {
              "op" : "==",
              "left" : {
                "type" : "field",
                "value" : ["udp", "$valid$"]
              },
              "right" : {
                "type" : "hexstr",
                "value" : "0x01"
              }
            }
          },
          "true_next" : "udp_check",
          "false_next" : "node_11"
        },
        {
          "name" : "node_11",
          "id" : 2,
          "source_info" : {
            "filename" : "test_data/07-MultiProtocol.p4",
            "line" : 309,
            "column" : 13,
            "source_fragment" : "valid(icmp)"
          },
          "expression" : {
            "type" : "expression",
            "value" : {
              "op" : "==",
              "left" : {
                "type" : "field",
                "value" : ["icmp", "$valid$"]
              },
              "right" : {
                "type" : "hexstr",
                "value" : "0x01"
              }
            }
          },
          "true_next" : "icmp_check",
          "false_next" : "set_egress"
        }
      ]
    },
    {
      "name" : "egress",
      "id" : 1,
      "init_table" : null,
      "tables" : [],
      "action_profiles" : [],
      "conditionals" : []
    }
  ],
  "checksums" : [],
  "force_arith" : [],
  "extern_instances" : [],
  "field_aliases" : [
    [
      "queueing_metadata.enq_timestamp",
      ["standard_metadata", "enq_timestamp"]
    ],
    [
      "queueing_metadata.enq_qdepth",
      ["standard_metadata", "enq_qdepth"]
    ],
    [
      "queueing_metadata.deq_timedelta",
      ["standard_metadata", "deq_timedelta"]
    ],
    [
      "queueing_metadata.deq_qdepth",
      ["standard_metadata", "deq_qdepth"]
    ],
    [
      "intrinsic_metadata.ingress_global_timestamp",
      ["standard_metadata", "ingress_global_timestamp"]
    ],
    [
      "intrinsic_metadata.lf_field_list",
      ["standard_metadata", "lf_field_list"]
    ],
    [
      "intrinsic_metadata.mcast_grp",
      ["standard_metadata", "mcast_grp"]
    ],
    [
      "intrinsic_metadata.resubmit_flag",
      ["standard_metadata", "resubmit_flag"]
    ],
    [
      "intrinsic_metadata.egress_rid",
      ["standard_metadata", "egress_rid"]
    ]
  ]
}