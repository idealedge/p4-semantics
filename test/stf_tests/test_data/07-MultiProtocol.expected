10101010101010101010101010101010101010101010101010111011101110111011101110111011101110111011101100001000000000000100010100000000000000000010110000000000000000000000000001100101010000000001000101101100010101010000000100000010000000110000010000001010000000000000000000000010000000000110010000000000011001010000000000010000000000000000000000000000000000010000001000000011000001000000010100000110000001110000100000001001000010100000101100001100000011010000111000001111 3